crates/factor-otel/src/host.rs

```diff
@@ -56,3 +56,25 @@ impl wasi_otel::Host for InstanceState {
             .into())
     }
 }
+
+impl wasi::otel::metrics::Host for InstanceState {
+    async fn export(
+        &mut self,
+        metrics: wasi::otel::metrics::ResourceMetrics,
+    ) -> spin_core::wasmtime::Result<std::result::Result<(), wasi::otel::metrics::Error>> {
+        let mut rm: opentelemetry_sdk::metrics::data::ResourceMetrics = metrics.into();
+        match self.metric_exporter.export(&mut rm).await {
+            Ok(_) => Ok(Ok(())),
+            Err(e) => match e {
+                OTelSdkError::AlreadyShutdown => {
+                    Ok(Err("Shutdown has already been invoked".to_string()))
+                }
+                OTelSdkError::InternalFailure(e) => Ok(Err("Internal failure: ".to_string() + &e)),
+                OTelSdkError::Timeout(d) => Ok(Err(format!(
+                    "Operation timed out after {} seconds",
+                    d.as_secs()
```

Comment:
Probably fine for now, but we'll need to think about the errors we expose to the guest. This strikes me as too much detail.